hack/ci-e2e.sh

```diff
@@ -59,6 +59,108 @@ export BOOT_MAC_ADDRESS
 # These are the VBMC defaults (used since we did not specify anything else for `vbmc add`).
 export BMC_USER=admin
 export BMC_PASSWORD=password
+# This is the IP address of the host where the image server is running.
+IMAGE_FILE="cirros-0.5.2-x86_64-disk.img"
+export IMAGE_MD5SUM="b874c39491a2377b8490f5f1e89761a4"
+IMAGE_IP_ADDRESS=$(echo $BMC_ADDRESS | sed -n 's|ipmi://\([^:]*\).*|\1|p')
+export IMAGE_URL="http://${IMAGE_IP_ADDRESS}/${IMAGE_FILE}"
+
+## Setup image server
+# Create a directory for images
+mkdir -p images
+pushd images
+
+# Check if IMAGE_FILE already exists
+if [[ ! -f $IMAGE_FILE ]]; then
+    wget http://download.cirros-cloud.net/0.5.2/$IMAGE_FILE
+else
+    echo "$IMAGE_FILE already exists. Skipping download."
+fi
+
+popd
+
+# Check if container "image-server" is running
+if [ "$(docker ps -q -f name=image-server)" ]; then
+    echo "Container 'image-server' is already running. Removing..."
+    docker rm -f image-server
+fi
+
+# Run image server
+docker run --rm --name image-server -d -p 80:8080 -v "$(pwd)/images:/usr/share/nginx/html" nginxinc/nginx-unprivileged
+
+## Setup image server
```

Comment:
Everything from here down until line 163 looks like duplicate or older version of the same code. Probably some mistake when squashing?